cmd/prow-job-dispatcher/main.go

```diff
@@ -191,45 +191,62 @@ func (cv *clusterVolume) findClusterForJobConfig(cloudProvider string, jc *prowc
+				errs = append(errs, err)
+			}
 		}
 	}
 
 	for k := range jc.PostsubmitsStatic {
 		for _, job := range jc.PostsubmitsStatic[k] {
-			cv.addToVolume(rCloudProvider, cluster, job.JobBase, path, config, jobVolumes)
+			if err := cv.addToVolume(rCloudProvider, cluster, job.JobBase, path, config, jobVolumes); err != nil {
+				errs = append(errs, err)
+			}
 		}
 	}
 	for _, job := range jc.Periodics {
-		cv.addToVolume(rCloudProvider, cluster, job.JobBase, path, config, jobVolumes)
+		if err := cv.addToVolume(rCloudProvider, cluster, job.JobBase, path, config, jobVolumes); err != nil {
+			errs = append(errs, err)
+		}
 	}
 
 	cv.mutex.Unlock()
-	return cluster
+	return cluster, utilerrors.NewAggregate(errs)
 }
 
-func (cv *clusterVolume) addToVolume(cloudProvider, cluster string, jobBase prowconfig.JobBase, path string, config *dispatcher.Config, jobVolumes map[string]float64) {
-	determinedCluster, canBeRelocated := config.DetermineClusterForJob(jobBase, path)
+func (cv *clusterVolume) addToVolume(cloudProvider, cluster string, jobBase prowconfig.JobBase, path string, config *dispatcher.Config, jobVolumes map[string]float64) error {
+	determinedCluster, canBeRelocated, err := config.DetermineClusterForJob(jobBase, path)
+	if err != nil {
+		return fmt.Errorf("failed to determine cluster for the job %s in path %q: %w", jobBase.Name, path, err)
+	}
 	if cluster == string(determinedCluster) || canBeRelocated {
 		cv.clusterVolumeMap[cloudProvider][cluster] = cv.clusterVolumeMap[cloudProvider][cluster] + jobVolumes[jobBase.Name]
 	} else if determinedCloudProvider := config.IsInBuildFarm(determinedCluster); determinedCloudProvider != "" {
 		cv.clusterVolumeMap[string(determinedCloudProvider)][string(determinedCluster)] = cv.clusterVolumeMap[string(determinedCloudProvider)][string(determinedCluster)] + jobVolumes[jobBase.Name]
 	}
+	return nil
 }
 
 // dispatchJobConfig dispatches the jobs defined in a Prow jon config
-func (cv *clusterVolume) dispatchJobConfig(jc *prowconfig.JobConfig, path string, config *dispatcher.Config, jobVolumes map[string]float64) string {
+func (cv *clusterVolume) dispatchJobConfig(jc *prowconfig.JobConfig, path string, config *dispatcher.Config, jobVolumes map[string]float64) (string, error) {
 	cloudProvidersForE2ETests := getCloudProvidersForE2ETests(jc)
 	var cluster string
+	var err error
 	if cloudProvidersForE2ETests.Len() == 1 {
 		cloudProvider, _ := cloudProvidersForE2ETests.PopAny()
-		cluster = cv.findClusterForJobConfig(cloudProvider, jc, path, config, jobVolumes)
+		if cluster, err = cv.findClusterForJobConfig(cloudProvider, jc, path, config, jobVolumes); err != nil {
+			return "", fmt.Errorf("fail to find cluster for job config: %w", err)
+		}
+
 	} else {
-		cluster = cv.findClusterForJobConfig("", jc, path, config, jobVolumes)
+		if cluster, err = cv.findClusterForJobConfig("", jc, path, config, jobVolumes); err != nil {
+			return "", fmt.Errorf("fail to find cluster for job config: %w", err)
+		}
```

Comment:
Looks like this can be simplified even more to

```go
var cloudProvider string
if cloudProvidersForE2ETests.Len() == 1 {
  cloudProvider, _ = cloudProvidersForE2ETests.PopAny()
}
if cluster, err = cv.findClusterForJobConfig(cloudProvider, jc, path, config, jobVolumes); err != nil {
   return "", fmt.Errorf("fail to find cluster for job config: %w", err)
}
```
```

Reply:
Updated.